steam-client/src/main/java/com/avenga/steamclient/model/steam/SteamMessageCallback.java

```diff
@@ -1,20 +1,47 @@
 package com.avenga.steamclient.model.steam;
 
+import com.avenga.steamclient.base.GCPacketMessage;
+import com.avenga.steamclient.base.PacketMessage;
+import com.avenga.steamclient.steam.client.SteamClient;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 
 import java.util.concurrent.CompletableFuture;
 
+/**
+ * Callback wrapper for {@link SteamClient} to handle client {@link PacketMessage} and Game Coordinator {@link GCPacketMessage}
+ * message received from Steam Network.
+ *
+ * @param <TPacket> Type of the packet message received from Steam Network.
+ */
 @Getter
-@EqualsAndHashCode
-public class SteamMessageCallback<T> {
+@EqualsAndHashCode(callSuper = false)
+public class SteamMessageCallback<TPacket> implements CompletableCallback {
     private int sequence;
+    private int applicationId;
     private int messageCode;
-    private CompletableFuture<T> callback;
+    private CompletableFuture<TPacket> callback;
 
-    public SteamMessageCallback(int messageCode, int sequence) {
+
+    /**
+     * Creates steam callback wrapper for packet messages.
+     *
+     * @param messageCode Code of the packet message.
+     * @param applicationId Application id of the Steam client or games.
+     * @param sequence Sequence number of the callback.
+     * @param callback Completable callback which should handle correct packet message.
+     */
+    public SteamMessageCallback(int messageCode, int applicationId, int sequence, CompletableFuture<TPacket> callback) {
         this.sequence = sequence;
+        this.applicationId = applicationId;
         this.messageCode = messageCode;
-        this.callback = new CompletableFuture<>();
+        this.callback = callback;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void complete(Object genericPacketMessage) {
```

Comment:
Why not make CompletableCallback<T> interface with a method which accepts parameterized type T instead of casting Object to the parametrized class which declared in SteamMessageCallback?

Reply:
We have two different interfaces for packet messages:
- Regular Steam client - `PacketMessage`.
- Game Coordinator - `GCPacketMessage`.

We can't specify both of them in one BlockedQueue, because we need to specify exact type in such case. To overcome this limitation we need to wrap `CompletableFubure` like this and hide actual execution of the `complete` method in the wrapper with correct packet message interface.

Reply:
```java
public class SteamMessageCallback<TPacket> implements CompletableCallback<TPacket>

public void complete(TPacket genericPacketMessage)
```

Won't it working?

Reply:
Ooo
looked to the code more carefully
I understand what you mean